### PR TITLE
use adaptive_avg_pool2d instead of avg_pool2d

### LINKED ISSUE #56: New adaptive pooling layer.
It appears that models in the torchvision are now using adaptive poolings: `adaptive_avg_pool2d `
 to break the tie to the input size: [vision/densenet.py](https://github.com/pytorch/vision/blob/master/torchvision/models/densenet.py#L115)
 
Perhaps that simplify the constructor a little bit and generalize the usage even more?

### PR DESCRIPTION
**Pull request by aneeshnema**
Closes #56 
removed `self.avgpool_size` as no longer needed.
moved `scheduler.step()` below `train_epoch` to remove pytorch warning "UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`" when using pytorch version > 1.1.0.

### CODE REVIEW

**Comment by gpleiss** [comment]
Thanks @aneeshnema !
